herd/AArch64Sem.ml

```diff
@@ -299,6 +318,30 @@ module Make
                   read_mem_acquire_pc sz rd a ii)
           (read_reg_ord rs ii) ii
 
+      let movz sz rd k os ii =
+        let open AArch64Base in
+        let is_imm16 n = n >= 0 && n < 65535 in
+        assert (is_imm16 k);
+        begin match sz, os with
+        | V32, None | V64, None ->
+          (* Or'ing zero with value should zero out what's left *)
+          M.op Op.Or (V.intToV k) V.zero
+        | V32, Some(LSL(0|16 as s))
+        | V64, Some(LSL((0|16|32|48 as s))) ->
+          let v = V.op1 (Op.LeftShift s) (V.intToV k) in
+          M.op Op.Or v V.zero
+        | _, Some(LSL(n)) -> Warn.fatal
+                   "illegal shift immediate %d in %s instruction movz"
+                   n
+                   (pp_variant sz)
+        | _, Some(s) -> Warn.fatal
+                   "illegal shift operand %s in %s instruction movz"
+                   (pp_barrel_shift s pp_imm)
+                   (pp_variant sz)
```

Comment:
It may be clearer to users to print the wrong instruction (with AArch64.dump_instruction).

Reply:
good idea, will fix